packages/forms/src/UIForm/fields/NestedListView/NestedListView.utils.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Build the ListView component props to match current state's values
+ * @param {Array} items
+ * @param {Object} value
+ * @param {String} searchCriteria
+ * @returns {Array}
+ */
+export function getDisplayedItems(items, value, searchCriteria) {
+	let textFilter;
+	if (searchCriteria) {
+		textFilter = ({ label }) => label.toLowerCase().includes(searchCriteria.toLowerCase());
+	}
+
+	return (
+		items
+			.map(item => {
+				// Add boolean "checked" to checked children
```

Comment:
Please avoid all code commentaries they are :
- for 50%, useless (when duplicate the code)
- for 25%, pernicious (when one or the other change without the two)
- for 24%, just an indicator that the code is too complex or implicit or is  and need to be more simple and/or explicit.
- for 1%, an indication the code shouldn't be written like this, so the code mustn't exist like this and the comment not too :p

Here you even don't exactly do what the comment say, you don't add the value to the child, you create a new Array of wrappers with the child and an other value.

For adding semantic to code, breaking in named functions and variables are a good way. It adds an other benefit : the possibility to read the code and understand the _what_ is happening without the need to check exactly _how_

Reply:
So here it's a bit the 2)

Reply:
Source of your statistics ?
I don't think breaking down this code chunk into severals methods would be any better for understanding than commenting each part, that's completely subjective.
Those comments are giving an overview of what's done in the method without entering the details on how javascript would make his transformations.

Reply:
After re-reading, I just said something totally wrong about the first map, the comment was right with the code. my bad :/
(So it's no more the 2 because I'm a derp, but more the 1 :P, whatever...)